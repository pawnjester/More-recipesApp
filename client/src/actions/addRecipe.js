import axios from 'axios';
import { CREATE_RECIPE_SUCCESS, CREATE_RECIPE_FAILURE } from './types';

export const addRecipeSucess = newRecipe => ({
  type: CREATE_RECIPE_SUCCESS,
  newRecipe,
});

export const addRecipeFailure = error => ({
  type: CREATE_RECIPE_FAILURE,
  error,
});


const createRecipe = data => dispatch => axios
  .post('/api/v1/recipes', data)
  .then((response) => {
    dispatch(addRecipeSucess(response.data.recipe));
  })
  .catch((error) => {
    dispatch(addRecipeFailure(error.response.data.error));
  });

export default createRecipe;
