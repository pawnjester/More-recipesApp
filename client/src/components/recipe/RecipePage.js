import React, { Component } from 'react';
import RecipeForm from './RecipeForm';
import '../../styles/signup.scss';



class RecipePage extends Component {
  render() {
    return (
      <header id ="home-section">
      <div className= "dark-overlay">
      <div className= "home-inner">
        <div className ="container">
          <div className= "row justify-content-center">
            <div className='col-lg-6'>
              <div className= "card bg-white text-center card-form">
                <div className="card-body">
                  <h3>Please Add A Recipe</h3>
                  <RecipeForm />                  
                </div>
              </div>
            </div>
            
          </div>
          
        </div>
        
      </div>
      </div>
      
    </header>
    )
  }
}

export default RecipePage;
