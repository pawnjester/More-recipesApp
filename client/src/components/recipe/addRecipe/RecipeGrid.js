import React, { Component } from 'react';
import  Recipe  from './recipe';

export default class RecipeGrid extends Component {
  render() {
    return (
    <div className="col-md-4 col-xs-4">
        <Recipe />
    </div>
    )
  }
}
