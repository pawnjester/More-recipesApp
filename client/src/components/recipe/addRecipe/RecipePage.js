import React, { Component } from 'react';
import { connect } from 'react-redux';
import createRecipe from '../../../actions/recipeActions';
import getRecipes from '../../../actions/getRecipe';
import deleteRecipe from '../../../actions/deleteRecipe'
import { Button,Container, Modal, 
    ModalHeader, ModalBody, ModalFooter, Form, Label, Input, FormGroup, Col, FormText } from 'reactstrap';    
import PropTypes from 'prop-types';
import styles from '../../../styles/recipes.scss';
import Recipe from './recipe';
import NavigationBar from '../../NavigationBar';
import AddRecipeModal from '../../Modal/AddRecipeModal';

class RecipePage extends Component {

  constructor(props) {
    super(props);

    this.state = {
      modal:false,
      nestedModal:false
    };

    this.toggle = this.toggle.bind(this);
    this.onDeleteRecipe = this.onDeleteRecipe.bind(this);
  }

  componentWillMount() {
    // TODO: get recipes list
    this.props.getRecipes();
  }
  

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  onDeleteRecipe(recipeId) {
    this.props.deleteRecipe(recipeId)
  }
      
  render() {
    if (this.props.recipes === undefined) {
      return <h4>Please wait....</h4>
    }
    return (
      <div className="container text-center  ">
        <div className="heading">
          <h1 className="p-5 ">My Recipes</h1>
          <Button color="primary" className="addrecipebtn btn btn-outline-success " onClick = {this.toggle}> <i className="fa fa-plus"></i>Recipe</Button>
          <AddRecipeModal
            isOpen={this.state.modal}
            toggle={this.toggle}
            createRecipe ={this.props.createRecipe}
          /> 
        </div>
        <hr/>
        
        <div className="row high">
            {this.props.recipes.map((recipe) =>
              <Recipe recipe={recipe} key={recipe.id} deleteRecipe={this.onDeleteRecipe}/>
            )}
          </div>
        </div> 
     )
  }
}


RecipePage.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  deleteRecipe: PropTypes.func
}

const mapStateToProps = (state) => {
    return {
        recipes: state.recipeReducer.recipes
    }
}


export default connect(mapStateToProps, { createRecipe, getRecipes, deleteRecipe})(RecipePage);
