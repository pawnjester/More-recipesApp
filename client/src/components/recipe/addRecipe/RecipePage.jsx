import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button } from 'reactstrap';
import ReactPaginate from 'react-paginate';
import PropTypes from 'prop-types';
import Loader from '../../common/Loader';
import createRecipe from '../../../actions/addRecipe';
import getRecipes from '../../../actions/getRecipe';
import deleteRecipe from '../../../actions/deleteRecipe';
import getRecipeDetail from '../../../actions/getRecipeDetail';
import editRecipe from '../../../actions/editRecipe';
import NavigationBar from '../../NavigationBar';
import '../../../styles/recipes.scss';
import Recipe from './recipe';
import AddRecipeModal from '../../Modal/AddRecipeModal';
import Footer from '../../common/Footer';

/**
 * @description RecipePage component
 *
 * @class RecipePage
 *
 * @extends {Component}
 */
class RecipePage extends Component {
  /**
   * Creates an instance of RecipePage.
   *
   * @constructor
   *
   * @param {any} props
   *
   * @memberof RecipePage
   *
   * @returns {void}
   */
  constructor(props) {
    super(props);

    this.state = {
      modal: false,
      recipes: [],
      loading: true,
      currentPage: 1,
      pages: 1,
      errors: '',
    };

    this.toggle = this.toggle.bind(this);
    this.onDeleteRecipe = this.onDeleteRecipe.bind(this);
    this.onPageChange = this.onPageChange.bind(this);
  }
  /**
 * @description get recipes
 *
 * @method
 *
 * @memberof RecipePage
 *
 * @returns {void}
 */
  componentWillMount() {
    this.props.getRecipes()
      .then(() => {
        this.setState({ loading: false });
      });
  }
  /**
 * @description set state
 *
 * @param {any} nextProps
 *
 * @memberof RecipePage
 *
 * @returns {void}
 */
  componentWillReceiveProps(nextProps) {
    const {
      currentPage
    } = this.state;
    this.setState({
      recipes: nextProps.recipes,
      pages: nextProps.pages,
      errors: nextProps.errors,
    });
    if (nextProps.deleted) {
      this.props.getRecipes(nextProps.totalContent < 1 ?
        currentPage - 1 :
        currentPage);
    }
  }

  /**
 *
 * @param {any} recipeId
 *
 * @memberof RecipePage
 *
 * @returns {void}
 */
  onDeleteRecipe(recipeId) {
    this.props.deleteRecipe(recipeId);
  }
  /**
 *@description set state to currentPage
 *
 * @param {any} data
 *
 * @memberof RecipePage
 *
 * @returns {void}
 */
  onPageChange(data) {
    data.selected += 1;
    this.props.getRecipes(data.selected);
    this.setState({
      currentPage: data.selected
    });
  }
  /**
 *@description toggle modal
 *
 * @memberof RecipePage
 *
 * @returns {void}
 */
  toggle() {
    this.setState({
      modal: !this.state.modal,
    });
  }

  /**
 *
 * @description renders the jsx element
 *
 * @memberof RecipePage
 *
 * @returns {void}
 */
  render() {
    const {
      recipes, pages, currentPage
    } = this.state;
    const recipeList = recipes || [];

    return (
      <div>
        <NavigationBar search="true" />
        <div className="container text-center">
          <div className="heading">
            <h1 className="p-5 ">My Recipes</h1>
            <Button
              color="primary"
              className="addrecipebtn btn btn-outline-success "
              onClick={this.toggle}
            >
              <i className="fa fa-plus" />Recipe
            </Button>
            <AddRecipeModal
              isOpen={this.state.modal}
              toggle={this.toggle}
              createRecipe={this.props.createRecipe}
              errors={this.state.errors}
              getRecipe={this.props.getRecipes}
              currentPage={currentPage}
            />
          </div>
          <hr />
          <div className="row high">
            {recipeList && recipeList.length === 0 &&
              (<h4 className="mt-5 text-center no-recipes"> No recipes yet </h4>)}
            {this.state.loading ?
              <Loader Loading={this.state.loading} /> :
              recipeList &&
              recipeList.map(recipe =>
                (<Recipe
                  recipe={recipe}
                  key={recipe.id}
                  deleteRecipe={this.onDeleteRecipe}
                  getAllRecipes={this.props.getRecipes}
                />))}

          </div>
        </div>
        {recipeList.length > 0 && <ReactPaginate
          pageCount={pages}
          pageRangeDisplayed={5}
          marginPagesDisplayed={3}
          previousLabel="Previous"
          nextLabel="Next"
          breakClassName="text-center"
          initialPage={0}
          containerClassName="container pagination justify-content-center"
          pageClassName="page-item"
          pageLinkClassName="page-link"
          activeClassName="page-item active"
          previousClassName="page-item"
          nextClassName="page-item"
          nextLinkClassName="page-link"
          previousLinkClassName="page-link"
          onPageChange={this.onPageChange}
        />}

        <Footer />
      </div>
    );
  }
}


RecipePage.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  deleteRecipe: PropTypes.func.isRequired,
  getRecipes: PropTypes.func.isRequired,
  recipes: PropTypes.arrayOf(PropTypes.any).isRequired,
  errors: PropTypes.objectOf(PropTypes.any).isRequired,
  deleted: PropTypes.bool.isRequired,
  totalContent: PropTypes.number.isRequired
};

const mapStateToProps = state => ({
  recipes: state.recipeReducer.recipes,
  totalContent: state.recipeReducer.totalContent,
  pages: state.recipeReducer.pages,
  errors: state.recipeReducer.error,
  deleted: state.recipeReducer.deleted
});

export default connect(
  mapStateToProps,
  {
    createRecipe, getRecipes, deleteRecipe, getRecipeDetail, editRecipe,
  },
)(RecipePage);
