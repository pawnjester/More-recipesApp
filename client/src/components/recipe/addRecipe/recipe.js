import React, { Component } from 'react';

export default class Recipe extends Component {
  constructor(props) {
    super(props)
    this.onDelete = this.onDelete.bind(this);
  }

  onDelete =(event) => {
    console.log('we are deleting');                          
    this.props.deleteRecipe(this.props.recipe.id);
    console.log('we are deleting******');                
  }
  render() {
    return (
      <div className="col-md-4 col-xs-12 " >
      
        <div className="card">
            <img className="card-img-top" src= {this.props.recipe.imageUrl}/>
            <div className="card-body">
              <h4 className="card-title">{this.props.recipe.name}</h4>
              <p className="card-text">{this.props.recipe.ingredients}</p>
              <p className="card-text">{this.props.recipe.method}</p>

            </div>
            <div className="card-body clearfix">
              <div className="text-left text-success float-left">
                <i className="fa fa-pencil" aria-hidden="true"></i><a href="" data-toggle="modal" data-target="#editModal"><span id='clickableAwesomeFont' className="text-success">&nbsp;Edit</span></a>
              </div>
              <div className="text-right text-danger float-right">
                <i className="fa fa-trash" aria-hidden="true"></i><span id='clickableAwesomeFont' onClick = {() =>this.onDelete()}className="delete">&nbsp;Delete</span>
              </div>
              <div className="text-center text-primary float-center">
                <i className="fa fa-eye" aria-hidden="true"></i><span id='clickableAwesomeFont' className="view">&nbsp;View</span>
              </div>
            </div>
          </div>
          </div>
          
    )
  }
}
