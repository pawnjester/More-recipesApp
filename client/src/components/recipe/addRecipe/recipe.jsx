import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import EditRecipeModal from '../../Modal/EditRecipeModal';


class Recipe extends Component {
  constructor(props) {
    super(props);
    this.onDelete = this.onDelete.bind(this);

    this.state = {
      modal: false,
    };
    this.toggle = this.toggle.bind(this);
  }
  onDelete() {
    this.props.deleteRecipe(this.props.recipe.id);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal,
    });
  }

  render() {
    const style = {
      height: 200,
    };
    const { recipe } = this.props;
    return (
      <div className="col-md-4 col-xs-12 " >
        <div className="card">
          <img className="card-img-top" src={recipe.imageUrl} style={style} alt="recipeImage" />
          <div className="card-body">
            <h4 className="card-title">{recipe.name}</h4>
            <p className="card-text">{recipe.ingredients}</p>
            <p className="card-text">{recipe.method}</p>

          </div>
          <div className="card-body clearfix">
            <div className="text-left text-success float-left">
              <i className="fa fa-pencil" aria-hidden="true" />
              <span id="clickableAwesomeFont" onClick={this.toggle} >&nbsp;Edit</span>
              <EditRecipeModal
                recipe={recipe}
                isOpen={this.state.modal}
                toggle={this.toggle}
                // editRecipe={this.props.editRecipe}
                id={this.props.recipe.id}
                getAllRecipes={this.props.getAllRecipes}
              />
            </div>

            <div className="text-right text-danger float-right">
              <i className="fa fa-trash" aria-hidden="true" /><span id="clickableAwesomeFont" onClick={() => this.onDelete()} className="delete">&nbsp;Delete</span>
            </div>
            <Link to={`/detail/${recipe.id}`}>
              <div className="text-center text-primary float-center">
                <i className="fa fa-eye" aria-hidden="true" /><span id="clickableAwesomeFont" className="view" >&nbsp;View</span>
              </div>
            </Link>

          </div>
        </div>
      </div>
    );
  }
}

export default Recipe;
