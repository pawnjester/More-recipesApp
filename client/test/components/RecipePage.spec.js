import React from 'React';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import sinon from 'sinon';
import { provider } from 'react-redux';
import { RecipePage } from '../../src/components/recipe/addRecipe/RecipePage';
import store from '../../src/store/store';

configure({ adapter: new Adapter() });


const props = {
  createRecipe: jest.fn(),
  getRecipes: jest.fn(() => Promise.resolve()),
  deleteRecipe: jest.fn(),
  getRecipeDetail: jest.fn(),
  editRecipe: jest.fn(),
  recipes: [],
  errors: {},
  totalContent: 3,
  deleted: true
};
const setup = () => {
  const shallowWrapper = shallow(<RecipePage {...props}/> );
  return {
    shallowWrapper,
  }
};

describe('RecipePage Component', () => {
  it('should render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('onPageChange', () => {
  it('should work correctly', () => {
    sinon.spy(RecipePage.prototype, 'onPageChange');
    const props = {
      createRecipe: jest.fn(),
      getRecipes: jest.fn(() => Promise.resolve()),
      deleteRecipe: jest.fn(),
      getRecipeDetail: jest.fn(),
      editRecipe: jest.fn(),
      recipes: [],
      errors: {},
      totalContent: 3,
      deleted: true
    };
    const selected = {
      data: 1
    };
    const shallowWrapper = shallow(<RecipePage {...props} store={store} />);
    shallowWrapper.instance().onPageChange(selected);
    expect(shallowWrapper.instance().props.getRecipes).toBeCalled();
  });
});

describe('onDeleteRecipe should', () => {
  it('be called when delete button is clicked', () => {
    const { shallowWrapper } = setup();
    const recipeId = 1;
    shallowWrapper.instance().onDeleteRecipe(recipeId);
    expect(props.deleteRecipe.mock.calls.length).toEqual(1);
});
});

describe('toggle should', () => {
  it('set modal state', () => {
    const { shallowWrapper } = setup();
    shallowWrapper.setState({
      modal: true
    })
    shallowWrapper.instance().toggle();
});
});
