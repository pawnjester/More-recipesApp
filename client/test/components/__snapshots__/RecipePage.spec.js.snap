// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePage Component should render correctly 1`] = `
<div>
  <Connect(NavigationBar)
    search="true"
  />
  <div
    className="container text-center"
  >
    <div
      className="heading"
    >
      <h1
        className="p-5 "
      >
        My Recipes
      </h1>
      <Button
        className="addrecipebtn btn btn-outline-success "
        color="primary"
        onClick={[Function]}
        tag="button"
      >
        <i
          className="fa fa-plus"
        />
        Recipe
      </Button>
      <AddRecipeModal
        createRecipe={[MockFunction]}
        currentPage={1}
        errors=""
        getRecipe={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
          }
        }
        isOpen={false}
        serverError={Object {}}
        toggle={[Function]}
      />
    </div>
    <hr />
    <div
      className="high"
    >
      <Loader
        Loading={true}
      />
    </div>
  </div>
  <Footer />
</div>
`;
